tests/e2e/csi_snapshot_utils.go

```diff
@@ -0,0 +1,593 @@
+	framework.Logf("Volume snapshot name is : %s", volumeSnapshot.Name)
+
+	ginkgo.By("Verify volume snapshot is created")
+	volumeSnapshot, err = waitForVolumeSnapshotReadyToUse(*snapc, ctx, namespace, volumeSnapshot.Name)
+	if err != nil {
+		return nil, nil, false, false, err
+	}
+	snapshotCreated := true
+	if volumeSnapshot.Status.RestoreSize.Cmp(resource.MustParse(diskSize)) != 0 {
+		return nil, nil, false, false, fmt.Errorf("unexpected restore size")
+	}
+
+	ginkgo.By("Verify volume snapshot content is created")
+	snapshotContent, err := snapc.SnapshotV1().VolumeSnapshotContents().Get(ctx,
+		*volumeSnapshot.Status.BoundVolumeSnapshotContentName, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, false, false, err
+	}
+	snapshotContentCreated := true
+	if !*snapshotContent.Status.ReadyToUse {
+		return nil, nil, false, false, fmt.Errorf("volume snapshot content is not ready to use")
+	}
+
+	return volumeSnapshot, snapshotContent, snapshotCreated, snapshotContentCreated, nil
+}
+
+// getPersistentVolumeClaimSpecWithDatasource return the PersistentVolumeClaim
+// spec with specified storage class.
+func getPersistentVolumeClaimSpecWithDatasource(namespace string, ds string, storageclass *storagev1.StorageClass,
+	pvclaimlabels map[string]string, accessMode v1.PersistentVolumeAccessMode,
+	datasourceName string, snapshotapigroup string) *v1.PersistentVolumeClaim {
+	disksize := diskSize
+	if ds != "" {
+		disksize = ds
+	}
+	if accessMode == "" {
+		// If accessMode is not specified, set the default accessMode.
+		accessMode = v1.ReadWriteOnce
+	}
+	claim := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "pvc-",
+			Namespace:    namespace,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				accessMode,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): resource.MustParse(disksize),
+				},
+			},
+			StorageClassName: &(storageclass.Name),
+			DataSource: &v1.TypedLocalObjectReference{
+				APIGroup: &snapshotapigroup,
+				Kind:     "VolumeSnapshot",
+				Name:     datasourceName,
+			},
+		},
+	}
+
+	if pvclaimlabels != nil {
+		claim.Labels = pvclaimlabels
+	}
+
+	return claim
+}
+
+/*
+changeDeletionPolicyOfVolumeSnapshotContentOnGuest changes the deletion policy
+of volume snapshot content from delete to retain in Guest Cluster
+*/
+func changeDeletionPolicyOfVolumeSnapshotContentOnGuest(ctx context.Context,
+	snapshotContent *snapV1.VolumeSnapshotContent, snapc *snapclient.Clientset,
+	namespace string) (*snapV1.VolumeSnapshotContent, error) {
+
+	// Retrieve the latest version of the object
+	latestSnapshotContent, err := snapc.SnapshotV1().VolumeSnapshotContents().Get(ctx, snapshotContent.Name,
+		metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Apply changes to the latest version
+	latestSnapshotContent.Spec.DeletionPolicy = snapV1.VolumeSnapshotContentRetain
```

Comment:
Can we take this deletion policy as a param to this function, so that anyone can change it to delete or retain. Also if needed this can be used in vanilla as well.

Reply:
make sense, fixed